rmw_zenoh_cpp/src/detail/rmw_data_types.hpp

```diff
@@ -284,7 +282,7 @@ class ZenohReply final
 
   ~ZenohReply();
 
-  std::optional<z_sample_t> get_sample() const;
+  const z_loaned_sample_t * get_sample() const;
```

Comment:
Not a fan of this change as it requires the user to check if the returned value is a nullptr.

For consistency, it would be good to retain the `std::optional` return  or another alternative could be to update this function to  (similar to how `get_query()` works above)
```suggestion
  const z_loaned_reply_t * get_reply() const;
```

Then users can call `z_reply_ok()` on this loaned reply and handle the nullptr case accordingly.

Reply:
Included in 54dd96a. Now we consistently keep a pointer to owned data and provide a get function to access the loaned data. Users can fetch the sample and check its validity via `z_reply_ok`.